src/frontend/planner_test/tests/testdata/expr.yaml

```diff
@@ -146,11 +146,11 @@
     -- When it is invalid, PostgreSQL reports error during explain, but we have to wait until execution as of now.
```

Comment:
We don't have to wait till execution now. Could we add some tests to show these type of literal casting errors?

Reply:
Good catch. I should have updated the test cases.

However, we are actually still conservative here. This error is still reported during execution. As the comment in `new_cast` reads:
```
            // else when eager parsing fails, just proceed as normal.
            // Some callers are not ready to handle `'a'::int` error here.
```
I decided not to update all callers in this PR to avoid making it too long. Maybe it is not worth it handling here if we are going to add a dedicated and general const folding phase.